tests/system/src/test/java/test/robot/javafx/scene/PixelBufferDrawTest.java

```diff
@@ -0,0 +1,317 @@
+        sourceIntBuffer = createIntBuffer(IMAGE_WIDTH, IMAGE_HEIGHT / 2, isDirect, TEST_COLOR);
+        PixelFormat<IntBuffer> pixelFormat = PixelFormat.getIntArgbPreInstance();
+        intPixelBuffer = new PixelBuffer<>(IMAGE_WIDTH, IMAGE_HEIGHT, sharedBuffer, pixelFormat);
+    }
+
+    private ImageView createImageViewPB(PixelBuffer<? extends Buffer> pixelBuffer) {
+        return new ImageView(new WritableImage(pixelBuffer));
+    }
+
+    private void verifyColor(Color color1, Color color2) {
+        for (int i = 0; i < root.getChildren().size(); i++) {
+            final int x = (int) (scene.getWindow().getX() + scene.getX() +
+                    root.getChildren().get(i).getLayoutX() + IMAGE_WIDTH / 2);
+            final int y = (int) (scene.getWindow().getY() + scene.getY() +
+                    root.getChildren().get(i).getLayoutY() + IMAGE_HEIGHT * 0.45);
+            Util.runAndWait(() -> {
+                actualColor = robot.getPixelColor(x, y);
+            });
+            Assert.assertEquals(color1, actualColor);
+
+            final int x1 = (int) (scene.getWindow().getX() + scene.getX() +
+                    root.getChildren().get(i).getLayoutX() + IMAGE_WIDTH / 2);
+            final int y1 = (int) (scene.getWindow().getY() + scene.getY() +
+                    root.getChildren().get(i).getLayoutY() + IMAGE_HEIGHT * 0.55);
+            Util.runAndWait(() -> {
+                actualColor = robot.getPixelColor(x1, y1);
+            });
+            Assert.assertEquals(color2, actualColor);
+        }
+    }
+
+    private <T extends Buffer>void performTest(PixelBuffer<T> pixelBuffer, Callback<PixelBuffer<T>, Rectangle2D> callback) {
+        // Step #2
+        Util.runAndWait(() -> {
+            for (int i = 0; i < NUM_IMAGES; i++)
+            root.getChildren().add(createImageViewPB(pixelBuffer));
+        });
+        delay();
+        // Step #3
+        verifyColor(INIT_COLOR, INIT_COLOR);
+
+        // Step #4
+        Util.runAndWait(() -> {
+            pixelBuffer.updateBuffer(callback);
+        });
+        delay();
+
+        // Step #5
+        verifyColor(TEST_COLOR, INIT_COLOR);
+    }
+
+    @Test
+    public void testIntArgbPreDirectBuffer() {
+        // Step #1
+        createIntPixelBuffer(true);
+        performTest(intPixelBuffer, intBufferCallback);
+    }
+
+    @Test
+    public void testIntArgbPreIndirectBuffer() {
+        // Step #1
+        createIntPixelBuffer(false);
+        performTest(intPixelBuffer,intBufferCallback);
+    }
+
+    @Test
+    public void testByteBgraPreDirectBuffer() {
+        // Step #1
+        createBytePixelBuffer(true);
+        performTest(bytePixelBuffer, byteBufferCallback);
+    }
+
+    @Test
+    public void testByteBgraPreIndirectBuffer() {
+        // Step #1
+        createBytePixelBuffer(false);
+        performTest(bytePixelBuffer, byteBufferCallback);
+    }
+
```

Comment:
Can you also add a test that calls `PixelBuffer.updateBuffer` from the wrong thread and verify that you get `IllegalStateException`?

Reply:
I shall add this to unit test, `PixelBufferTest.java` file

Reply:
> I shall add this to unit test, `PixelBufferTest.java` file

No, that won't work. The graphics tests run with the StubToolkit which cannot test any threading issues.